ChameleonForms/PartialViewForm.cs

```diff
@@ -0,0 +1,49 @@
+﻿using System;
+using System.Linq.Expressions;
+using System.Web;
+using System.Web.Mvc;
+using ChameleonForms.FieldGenerators;
+using ChameleonForms.Templates;
+using ChameleonForms.Utils;
+
+namespace ChameleonForms
+{
+    internal class PartialViewForm<TModel, TPartialModel> : IForm<TPartialModel>

```

Comment:
This class is the most complex. I have to give a form that looks like the parent form, but writes to the text writer for the partial otherwise the output is out of order.